src/WindowManager.vala

```diff
@@ -2222,10 +2226,14 @@ namespace Gala {
 
             switch_workspace_with_gesture = false;
             animating_switch_workspace = false;
+
+            return true;
         }
 
         public override void kill_switch_workspace () {
-            end_switch_workspace ();
+            if (end_switch_workspace ()) {
+                switch_workspace_completed ();
+            }
```

Comment:
why do we have that here? it looks unrelated to the topic of this PR

Reply:
I'm sorry that change was actually a remnant from some testing I did. 
However the other change in end_switch_workspace () is required because now we can actually end up in a situation where windows and their parents is null. However just returning was wrong there (which wasn't noticed previously because we always had wingpanel and dock).

(FYI this will be completely removed in #2225 anyways)